src/service/standard-actions-impl.js

```diff
@@ -101,10 +116,14 @@ export class StandardActions {
    * @param {number=} opt_actionIndex
    * @param {!Array<!./action-impl.ActionInfoDef>=} opt_actionInfos
    * @return {?Promise}
-   * @throws {Error} If action is not recognized.
+   * @throws {Error} If action is not recognized or is not whitelisted.
    */
   handleAmpTarget(invocation, opt_actionIndex, opt_actionInfos) {
     const method = invocation.method;
+    if (this.ampActionWhitelist_ &&
+      !this.ampActionWhitelist_.includes(method)) {
+      throw user().createError('AMP action', method, 'is not whitelisted');
```

Comment:
`"AMP.print" is not whitelisted in <meta name="amp-action-whitelist" content="AMP.setState">. See (link) for more details."

(link) should point to a new section in amp-actions-and-events.md.

Reply:
Adding this meta tag makes the amp document invalid (as the insertion of meta tag is post validation). With this in mind, could you please clarify what should go in the documentation?

Reply:
No documentation update then. 😄 Just want to make sure the error message is developer friendly.

Reply:
@lannka This is the conversation I told you about